src/nimony/sem.nim

```diff
@@ -2515,6 +2524,141 @@ proc semBorrow(c: var SemContext; fn: StrId; beforeParams: int) =
   var it = Item(n: n, typ: c.types.autoType)
   semProcBody c, it
 
+proc getParamsType(c: var SemContext; paramsAt: int): seq[TypeCursor] =
```

Comment:
Why not simply traverse the parameter list after it has been semchecked?

Reply:
I don't know how to traverse the parameter list